ocaml/xapi/dbsync_slave.ml

```diff
@@ -242,7 +242,17 @@ let update_env __context sync_keys =
     );
 
   (* record who we are in xapi_globs *)
-  Xapi_globs.localhost_ref := Helpers.get_localhost ~__context;
+  let localhost = Helpers.get_localhost ~__context in
+  Xapi_globs.localhost_ref := localhost;
+
+  (** Normally the resident_on field would be set by the helper which creates
+      the task, but it uses this `localhost_ref` to do so, which we have only
+      just initialized above. Therefore we manually set it here *)
+  let set_context_resident_on () =
+    let task_ref = Context.get_task_id __context in
+    Db.Task.set_resident_on ~__context ~self:task_ref ~value:localhost;
+  in
+  set_context_resident_on ();
```

Comment:
You could just have the body of `set_context_resident_on` here if you call it immediately.

Reply:
i agree this is overkill here